### PR TITLE
✨ Sequences

### PR DESCRIPTION
**Pull request by RTLS**
✨ Sequences

### CODE REVIEW

**Comment by RTLS** [comment]
* **#392** <a href="https://app.graphite.dev/github/pr/sequinstream/sequin/392?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @RTLS and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: assets/svelte/sequences/Index.svelte [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+              <Popover>
+                <PopoverTrigger>
+                  <Button
+                    variant="destructive"
+                    size="sm"
+                    on:click={() => handleDelete(sequence.id)}
+                    disabled={sequence.consumer_count > 0}
+                  >
+                    Delete
+                  </Button>
```
Comment (acco) [inline assets/svelte/sequences/Index.svelte:244]: Use `...` instead of delete button

File: lib/sequin/databases/databases.ex [hunk 2]
```diff
@@ -130,6 +132,43 @@ defmodule Sequin.Databases do
...
+    Repo.delete(sequence)
+  end
+
+  def delete_sequences(%PostgresDatabase{} = db) do
+    Sequence
+    |> Repo.delete_all(where: [postgres_database_id: db.id])
+    |> case do
+      {num_deleted, nil} -> {:ok, num_deleted}
+    end
+  end
```
Comment (acco) [inline lib/sequin/databases/databases.ex:170]: check to see if any related entities (consumer)

File: lib/sequin/databases/sequence.ex [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+  import Ecto.Changeset
+  import Ecto.Query
+
+  alias Sequin.Databases.PostgresDatabase
+
+  typed_schema "sequences" do
+    field :table_oid, :integer
+    field :table_schema, :string
+    field :table_name, :string
```
Comment (acco) [inline lib/sequin/databases/sequence.ex:13]: we'll update during Databases.update_tables

File: lib/sequin_web/live/sequences/index.ex [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+
+  defp apply_action(socket, :new, _params) do
+    socket
+    |> assign(:page_title, "New Sequence")
+    |> assign(:sequence, %Databases.Sequence{})
+  end
+
+  defp apply_action(socket, :index, _params) do
+    socket
+    |> assign(:page_title, "Listing Sequences")
```
Comment (acco) [inline lib/sequin_web/live/sequences/index.ex:48]:
```suggestion
    |> assign(:page_title, "Sequences")
```

File: lib/sequin_web/live/sequences/index.ex [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+
+  def handle_event("form_submitted", %{"form" => sequence_params}, socket) do
+    account_id = current_account_id(socket)
+
+    # It's okay if this raises on 404, because the form doesn't let you select a database that doesn't exist
+    {:ok, database} = Databases.get_db_for_account(account_id, sequence_params["postgres_database_id"])
+
+    # Find the selected table and column based on the OIDs
+    selected_table = Enum.find(database.tables, &(&1.oid == sequence_params["table_oid"]))
+    selected_column = Enum.find(selected_table.columns, &(&1.attnum == sequence_params["sort_column_attnum"]))
```
Comment (acco) [inline lib/sequin_web/live/sequences/index.ex:86]:
```suggestion
    selected_table = Sequin.Enum.find!(database.tables, &(&1.oid == sequence_params["table_oid"]))
    selected_column = Sequin.Enum.find!(selected_table.columns, &(&1.attnum == sequence_params["sort_column_attnum"]))
```

File: lib/sequin_web/live/sequences/index.ex [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+
+    # Update sequence_params with the required fields
+    sequence_params =
+      sequence_params
+      |> Map.put("postgres_database_id", database.id)
+      |> Map.put("table_schema", selected_table.schema)
+      |> Map.put("table_name", selected_table.name)
+      |> Map.put("sort_column_name", selected_column.name)
```
Comment (acco) [inline lib/sequin_web/live/sequences/index.ex:94]: after lifecycle, refactor?

File: assets/svelte/sequences/Form.svelte [hunk 7]
```diff
@@ -0,0 +1,252 @@
...
+  bind:showConfirmDialog
+  on:close={handleClose}
+>
+  <form on:submit={handleSubmit} class="space-y-6 max-w-4xl mx-auto mt-6">
+    <Card>
+      <CardHeader>
+        <CardTitle>Sequence Details</CardTitle>
+      </CardHeader>
+      <CardContent class="space-y-6">
+        <div class="space-y-2">
```
[the remark this answers is not in the document]
Comment (RTLS) [reply assets/svelte/sequences/Form.svelte:120]: no

**Pull request merged**